
Some SCSI-IDE cdrom drives take up to 15 seconds to respond following
an abort being issued to them. This patch changes ipr to only send
a cancel all to a device as part of request sense processing when
the device is running tagged command queueing.


---

 linux-2.6.7-rc2-bjking1/drivers/scsi/ipr.c |    5 +++++
 1 files changed, 5 insertions(+)

diff -puN drivers/scsi/ipr.c~ipr_only_tcq_cancel_all drivers/scsi/ipr.c
--- linux-2.6.7-rc2/drivers/scsi/ipr.c~ipr_only_tcq_cancel_all	2004-06-07 17:11:05.000000000 -0500
+++ linux-2.6.7-rc2-bjking1/drivers/scsi/ipr.c	2004-06-07 17:11:05.000000000 -0500
@@ -3535,6 +3535,11 @@ static void ipr_erp_cancel_all(struct ip
 
 	ipr_reinit_ipr_cmnd_for_erp(ipr_cmd);
 
+	if (!res->tcq_active) {
+		ipr_erp_request_sense(ipr_cmd);
+		return;
+	}
+
 	cmd_pkt = &ipr_cmd->ioarcb.cmd_pkt;
 	cmd_pkt->request_type = IPR_RQTYPE_IOACMD;
 	cmd_pkt->cdb[0] = IPR_CANCEL_ALL_REQUESTS;

_
